lasagne/regularization.py

```diff
@@ -1,15 +1,47 @@
 import theano.tensor as T
+from .layers import Layer, get_all_params
 
-from . import layers
 
+def l1(param):
+    """ L1 weight decay penalty """
+    return T.sum(abs(param))
 
-def l2(layer, include_biases=False):
-    if include_biases:
-        all_params = layers.get_all_params(layer)
-    else:
-        all_params = layers.get_all_non_bias_params(layer)
 
-    return sum(T.sum(p**2) for p in all_params)
+def l2(param):
+    """ L2 weight decay penalty """
+    return T.sum(param**2)
 
 
-# TODO: sparsity regularization
+def apply_penalty(params, penalty, **kwargs):
+    try:
+        if len(params) == 0:
+            return 0
+        else:
+            return sum(penalty(p, **kwargs) for p in params)
+    except TypeError:
+        return penalty(params, **kwargs)
+
+
+def regularize_layer_params(layer, penalty,
+                            tags={'regularizable': True}, **kwargs):
+    layers = [layer, ] if isinstance(layer, Layer) else layer
+    all_params = []
+
+    for layer in layers:
+        all_params += layer.get_params(**tags)
```

Comment:
Are we anticipating many penalty functions with keyword arguments to control their behaviour? Otherwise it would probably be nicer to replace the `**kwargs` for `regularize_layer_params` with `**tags`, in line with the other methods and functions that work with tags.